app/controllers/tickets_controller.rb

```diff
@@ -106,4 +106,189 @@ def edit_person_validators
       dob: dob_validation_issues,
     }
   end
+
+  private def get_user_and_person
+    user_id = params[:userId]
+    wca_id = params[:wcaId]
+
+    if user_id && !wca_id
+      user = User.find(user_id)
+      person = user.person
+    elsif !user_id && wca_id
+      person = Person.find_by(wca_id: wca_id)
+      user = person.user
+    elsif user_id && wca_id
+      person = Person.find_by(wca_id: wca_id)
+      user = User.find(user_id)
+    end
+    [user, person]
+  end
+
+  private def check_errors(user, person)
+    if user.present? && person.present? && person&.user != user
+      render status: :unprocessable_entity, json: {
+        error: "Person and user not linked.",
+      }
+      return true
+    end
+
+    if user.nil? && person.nil?
+      render status: :unprocessable_entity, json: {
+        error: "User ID and WCA ID is not provided.",
+      }
+      return true
+    end
+    false
+  end
+
+  def details_before_anonymization
+    user, person = get_user_and_person
+    return if check_errors(user, person)
+
+    anonymization_checks = {}
+    message_args = {}
+    action_items = []
+    non_action_items = []
+
+    # 1. Check whether the user is currently banned or not.
+    anonymization_checks[:user_currently_banned] = user&.banned?
+
+    # 2. Check whether the user was banned in the past.
+    anonymization_checks[:user_banned_in_past] = user&.banned_in_past?
```

Comment:
You can do this directly while creating the object above

```suggestion
    anonymization_checks = {
      user_currently_banned: user&.banned?,
      user_banned_in_past: user&.banned_in_past?,
    }
```